diff/diff_test.go

```diff
@@ -562,7 +552,7 @@ func TestParseFileDiffAndPrintFileDiff(t *testing.T) {
 			t.Fatal(err)
 		}
 		diff, err := ParseFileDiff(diffData)
-		if !reflect.DeepEqual(err, test.wantParseErr) {
+		if !cmp.Equal(err, test.wantParseErr) {
```

Comment:
this currently breaks travis CI. You have two options to fix it:

1. (preferred) Use `errors.Is` instead.
2. Pass https://godoc.org/github.com/google/go-cmp/cmp/cmpopts#IgnoreUnexported or https://godoc.org/github.com/google/go-cmp/cmp/cmpopts#EquateErrors as an option to `cmp.Equal`

Reply:
Having problem with each option:
1 `!errors.Is(err, test.wantParseErr)`
2.1 `!cmp.Equal(err, test.wantParseErr, cmpopts.IgnoreUnexported())`
2.2 `!cmp.Equal(err, test.wantParseErr, cmpopts.IgnoreUnexported(), cmpopts.EquateErrors())`
2.3 `!cmp.Equal(err, test.wantParseErr, cmpopts.EquateErrors())`

I guess it couldn't properly compare it as error, when error is hidden in `err` inside `ParseError` structure (failing example `ParseError{0, 0, ErrExtendedHeadersEOF}`). 

Changed back to `!reflect.DeepEqual()` for now.

Reply:
After discussion offline, removed pointer from `ParseError` to compare errors with `errors.Is(myErr, targetErr)`.